File: src/main/java/chapter5/example/HttpGetExample.java
package chapter5.example;

import common.Log;
import okhttp3.*;

import java.io.IOException;


public class HttpGetExample {

    private final OkHttpClient client = new OkHttpClient();

    private static final String URL_README =
            "https://raw.githubusercontent.com/yudong80/reactivejava/master/README.md";

    public void run() {
        Request request = new Request.Builder()
                .url(URL_README)
                .build();

        client.newCall(request).enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {
                e.printStackTrace();
            }

            @Override
            public void onResponse(Call call, Response response) throws IOException {
                Log.it(response.body().string());
            }
        });
    }

    public static void main(String[] args) {
        HttpGetExample demo = new HttpGetExample();
        demo.run();
    }

}

File: src/main/kotlin/chapter5/example/HttpGetExample_.kt
package chapter5.example

import common.Log_
import okhttp3.*
import java.io.IOException


class HttpGetExample_ {

    private val client = OkHttpClient()

    private val URL_README = "https://raw.githubusercontent.com/yudong80/reactivejava/master/README.md"

    fun run() {
        val request = Request.Builder()
                .url(URL_README)
                .build()

        client.newCall(request).enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {
                e.printStackTrace()
            }

            @Throws(IOException::class)
            override fun onResponse(call: Call, response: Response) {
                Log_.it(response.body().string())
            }
        })
    }

}


fun main(args: Array<String>) {
    val demo = HttpGetExample_()
    demo.run()
}
